pkg/eval/platform/platform.go

```diff
@@ -35,9 +37,46 @@ import (
 // Whether or not the platform is Microsoft Windows.
 // This is read-only.
 
+//elvdoc:fn hostname &omit-domain=false
+//
+// The host name of the system. If option `&omit-domain` is used any domain
+// components are stripped from the host name. If an error occurs while
+// fetching the host name `$nil` is substituted. This is implemented using
+// Go's [`os.Hostname()`](https://golang.org/pkg/os/#Hostname) function.
+
+//elvdoc:var machname
+//
+// The machine name; i.e., the first component of the hostname.
+// If an error occurs while fetching the machine name `$nil` is substituted.
+// This is implemented using Go's
+// [`os.Hostname()`](https://golang.org/pkg/os/#Hostname) function.
+// This is read-only.
+
+var osHostname = os.Hostname // to allow mocking in unit tests
+
+type hostnameOpt struct{ OmitDomain bool }
+
+func (o *hostnameOpt) SetDefaultOptions() { o.OmitDomain = false }
+
+func hostname(opts hostnameOpt) interface{} {
+	hostname, err := osHostname()
+	if err != nil {
+		return nil
+	}
+	if !opts.OmitDomain {
+		return hostname
+	}
+	comps := strings.SplitN(hostname, ".", 2)
+	return comps[0]
+}
+
 var Ns = eval.Ns{
 	"arch":       vars.NewReadOnly(runtime.GOARCH),
 	"os":         vars.NewReadOnly(runtime.GOOS),
 	"is-unix":    vars.NewReadOnly(isUnix),
 	"is-windows": vars.NewReadOnly(isWindows),
+}.AddGoFns("platform:", fns)
```

Comment:
Just inline the `fns` map here.

Reply:
I'm not a fan of that style from a readability standpoint. Especially as more functions get added in the future. But, okay, done.